src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int DEFAULT_THRESHOLD = 12;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = DEFAULT_THRESHOLD;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        if (table[hash % table.length] == null) {
+            table[hash % table.length] = new Node<>(hash, key, value, null);
+        } else {
+            Node<K, V> currentNode = table[hash % table.length];
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                size--;
```

Comment:
why size-- here?

let's remove size-- and just put size++ where needed (and not in all cases)